core/templates/dev/head/filtersSpec.js

```diff
@@ -264,4 +265,84 @@ describe('Testing filters', function() {
     expect(filter('please do not test empty string', 100)).toEqual(
       'Please do not test empty string');
   }));
+
+  it('should strip out all tags unless they contain the RTE tags',
```

Comment:
unless they contain the RTE tags --> except for img tags containing the whitelisted classes

(this filter should not "know" about the RTE)